networkx/algorithms/bidirectional.py

```diff
@@ -0,0 +1,508 @@
+# -*- coding: utf-8 -*-
+#    Copyright (C) 2015 by
+#    André Dietrich <dietrich@ivs.cs.uni-magdeburg.de>
+#    Sebastian Zug <zug@ivs.cs.uni-magdeburg.de>
+#    All rights reserved.
+#    BSD license.
+import networkx as nx
+
+__all__ = ['all_simple_paths',
+           'all_shortest_paths',
+           'has_path']
+
+
+def all_simple_paths(G, source, target, cutoff=None):
+    """Generate all simple paths in graph G from source to target.
+
+    A simple path is a path with no repeated nodes.
+
+    The result paths are returned ordered by there length, starting
+    from the shortest paths.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+
+    source : node
+       Starting node for path
+
+    target : node
+       Ending node for path
+
+    cutoff : integer, optional
+        Depth to stop the search. Only paths of length <= cutoff are returned.
+
+    Returns
+    -------
+    path_generator: generator
+       A generator that produces lists of simple paths, ordered by their
+       length starting from the shortes path. If there are no paths between
+       the source and target within the given cutoff the generator
+       produces no output.
+
+    Examples
+    --------
+    >>> import networkx as nx
+    >>> G = nx.complete_graph(4)
+    >>> for path in nx.bidirectional.all_simple_paths(G, source=0, target=3):
+    ...     print(path)
+    ...
+    [0, 3]
+    [0, 1, 3]
+    [0, 2, 3]
+    [0, 2, 1, 3]
+    [0, 1, 2, 3]
```

Comment:
This is causing a doctest failure on travis.
```
Expected:
    [0, 3]
    [0, 1, 3]
    [0, 2, 3]
    [0, 2, 1, 3]
    [0, 1, 2, 3]
Got:
    [0, 3]
    [0, 1, 3]
    [0, 2, 3]
    [0, 1, 2, 3]
    [0, 2, 1, 3]
```
Consider switching last two lists. Further, after a review note in the PR, instead of making a new commit for every change, you should try other options like `amend` or `squash`.